### PR TITLE
[READY]Makes a singuloproof var for areas, singuloproofs bubblegum shuttle arena and CTF.

### PR DESCRIPTION
**Pull request by silicons**
#Kinda bad as it is on zlevel 2, and can be used to completely  fuck over anyone entering it.
as for CTF, well, it's for ghosts, why would you break it?

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Does this prevent ANY BoH singulos from being made on z2? If so, 👎

**Comment by kevinz000** [comment]
@PKPenguin321 no, and I'm retesting this as apparently it's not on Z2?

**Comment by Kromgar** [comment]
Fixes #24089

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -71,10 +71,19 @@
...
 		user << "<span class='danger'>The Bluespace interfaces of the two devices catastrophically malfunction!</span>"
 		qdel(W)
-		var/obj/singularity/singulo = new /obj/singularity (get_turf(src))
-		singulo.energy = 300 //should make it a bit bigger~
-		message_admins("[key_name_admin(user)] detonated a bag of holding")
-		log_game("[key_name(user)] detonated a bag of holding")
+		var/turf/T = get_turf(src)
+		var/area/A = T.loc
+		var/singulod = FALSE
+		if(istype(A, /area/arena))
```
Comment (coiax) [inline code/game/objects/items/weapons/storage/backpack.dm]: I do not approve of this special casing in this manner.

If we want to make specific areas immune to the sing, just make a flag or a var in /area and have the sing check it on New() and in process()

Comment (kevinz000) [inline code/game/objects/items/weapons/storage/backpack.dm]: eehhh ok.

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 2]
```diff
@@ -71,10 +71,18 @@
...
 		qdel(W)
-		var/obj/singularity/singulo = new /obj/singularity (get_turf(src))
-		singulo.energy = 300 //should make it a bit bigger~
-		message_admins("[key_name_admin(user)] detonated a bag of holding")
-		log_game("[key_name(user)] detonated a bag of holding")
+		var/area/A = get_area(src)
+		var/singulod = FALSE
+		if(A.singuloproof)
+			user << "<span class='userdanger'>The bag of holding detonates from having its bluespace fields inhibited by the local area!</span>"
+			explosion(T, 2, 4, 8, 16)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/weapons/storage/backpack.dm]: owo what's this?

**Comment by Kromgar** [comment]
Wait why the fuck does it make the bags explode? Just make it not work

**Comment by KorPhaeron** [comment]
This is the wrong fix for the issue

**Pull request closed without merging**